src/Illuminate/Database/Query/Processors/SqlServerProcessor.php

```diff
@@ -17,9 +17,17 @@ class SqlServerProcessor extends Processor
      */
     public function processInsertGetId(Builder $query, $sql, $values, $sequence = null)
     {
-        $query->getConnection()->insert($sql, $values);
-
-        $id = $query->getConnection()->getPdo()->lastInsertId();
+        $connection = $query->getConnection();
+        $connection->insert($sql, $values);
+        if ($connection->getConfig('odbc') === true) {
+            $result = $connection->select('SELECT CAST(COALESCE(SCOPE_IDENTITY(), @@IDENTITY) AS int) AS insertid');
+            if (! $result) {
+                throw new Exception('Error retrieving lastInsertId');
```

Comment:
Thanks, I've fixed it up now.
Is there a standard I should follow for the Exception message?